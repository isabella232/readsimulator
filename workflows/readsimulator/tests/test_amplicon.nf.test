nextflow_workflow {

    name "Test workflow: READSIMULATOR"
    script "../main.nf"
    workflow "READSIMULATOR"
    tag "workflows"
    tag "readsimulator"
    tag "readsimulator_test_amplicon"

    test("amplicon = true") {

        when {
            workflow {
                """
                input[0] = Channel.of(
                    [ [ id:'first', seed:40] ],
                    [ [ id:'second', seed:41] ],
                    [ [ id:'third', seed:42] ],
                )
                """
            }
            params {
                amplicon                    = true
                amplicon_fw_primer          = 'AAAATAAT'
                amplicon_rv_primer          = 'GATTACTTT'
                amplicon_read_count         = 1000
                amplicon_crabs_ispcr_error  = 0
                fasta                       = 'https://raw.githubusercontent.com/nf-core/test-datasets/readsimulator/testdata/GCF_024334085.1_ASM2433408v1_genomic.fna.gz'
                outdir                      = "results"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.get(0).findAll { it != "multiqc_report" }
                ).match() }
            )
        }
    }
}
