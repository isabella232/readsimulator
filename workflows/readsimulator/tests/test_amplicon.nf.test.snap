{
    "amplicon = true": {
        "content": [
            [
                [
                    {
                        "id": "first",
                        "seed": 40,
                        "outdir": "art_illumina",
                        "datatype": "amplicon_illumina"
                    },
                    [
                        "first1.fq.gz:md5,5dfb2de5a3d547b4a1cbdfe8fe33eb78",
                        "first2.fq.gz:md5,3ff099ec08c9d1f1f87f7bcd1455aad3"
                    ]
                ],
                [
                    {
                        "id": "second",
                        "seed": 41,
                        "outdir": "art_illumina",
                        "datatype": "amplicon_illumina"
                    },
                    [
                        "second1.fq.gz:md5,fdb32a77b5d100efd77f9b253475b354",
                        "second2.fq.gz:md5,97b1101a3a2a3814f301ccd90b5c2f30"
                    ]
                ],
                [
                    {
                        "id": "third",
                        "seed": 42,
                        "outdir": "art_illumina",
                        "datatype": "amplicon_illumina"
                    },
                    [
                        "third1.fq.gz:md5,62ab3b38160c81c9d9fa972871f896d3",
                        "third2.fq.gz:md5,912ae27dc22fea5b3d2f8f3a5dd4fa08"
                    ]
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T14:38:49.385639018"
    }
}